tests/common/widgets/hits-per-page/options.ts

```diff
@@ -0,0 +1,175 @@
+import { createSearchClient } from '@instantsearch/mocks';
+import { normalizeSnapshot, wait } from '@instantsearch/testutils';
+import { screen } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+
+import type { HitsPerPageWidgetSetup } from '.';
+import type { TestOptions } from '../../common';
+
+export function createOptionsTests(
+  setup: HitsPerPageWidgetSetup,
+  { act }: Required<TestOptions>
+) {
+  describe('options', () => {
+    test('renders with required props', async () => {
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+        },
+        widgetParams: {
+          items: [
+            { label: '10', value: 10, default: true },
+            { label: '20', value: 20 },
+            { label: '30', value: 30 },
+          ],
+        },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
```

Comment:
I wonder if this should be combined with a hits widget so the length can be checked (although that will mostly rely on the search client mock don't think it's worth it)